File: backend/src/main/java/com/ccadmin/app/video/model/dto/VideoDetailDto.java
package com.ccadmin.app.video.model.dto;

import com.ccadmin.app.video.model.entity.ActorEntity;
import com.ccadmin.app.video.model.entity.TagEntity;
import com.ccadmin.app.video.model.entity.VideoCaptureEntity;
import com.ccadmin.app.video.model.entity.VideoCategoryEntity;
import com.ccadmin.app.video.model.entity.VideoEntity;
import java.util.List;

public class VideoDetailDto {
    public VideoEntity Video;
    public List<VideoCategoryEntity> Categories;
    public List<ActorEntity> Actors;
    public List<TagEntity> Tags;
    public List<VideoCaptureEntity> Captures;

    public VideoDetailDto(VideoEntity video, List<VideoCategoryEntity> categories, List<ActorEntity> actors, List<TagEntity> tags, List<VideoCaptureEntity> captures) {
        this.Video = video;
        this.Categories = categories;
        this.Actors = actors;
        this.Tags = tags;
        this.Captures = captures;
    }
}

File: frontend/src/app/enterprise/video/model/dto/VideoDetailDto.ts
import { ActorEntity } from '../entity/ActorEntity';
import { TagEntity } from '../entity/TagEntity';
import { VideoCaptureEntity } from '../entity/VideoCaptureEntity';
import { VideoCategoryEntity } from '../entity/VideoCategoryEntity';
import { VideoEntity } from '../entity/VideoEntity';

export class VideoDetailDto {
  Video: VideoEntity = new VideoEntity();
  Categories: VideoCategoryEntity[] = [];
  Actors: ActorEntity[] = [];
  Tags: TagEntity[] = [];
  Captures: VideoCaptureEntity[] = [];
}
